exchanges/exchange.go

```diff
@@ -1876,3 +1874,67 @@ func (b *Base) ParallelChanOp(channels []subscription.Subscription, m func([]sub
 
 	return errs
 }
+
+// Bootstrap function allows for exchange authors to supplement or override common startup actions
+// If exchange.Bootstrap returns false or error it will not perfom any other actions.
+// If it returns true, or is not implemented by the exchange, it will:
+// * Print debug startup information
+// * UpdateOrderExecutionLimits
+// * UpdateTradablePairs
+func Bootstrap(ctx context.Context, b IBotExchange) error {
+	if continueBootstrap, err := b.Bootstrap(ctx); !continueBootstrap || err != nil {
+		return err
+	}
+
+	if b.IsVerbose() {
+		wsURL := ""
+		wsEnabled := false
+		if w, err := b.GetWebsocket(); err != nil {
+			wsURL = w.GetWebsocketURL()
+			wsEnabled = w.IsEnabled()
+		}
+		log.Debugf(log.ExchangeSys, "%s Websocket: %s. (url: %s).\n", b.GetName(), common.IsEnabled(wsEnabled), wsURL)
+		b.PrintEnabledPairs()
+	}
+
+	if b.GetEnabledFeatures().AutoPairUpdates {
+		if err := b.UpdateTradablePairs(ctx, false); err != nil {
+			return fmt.Errorf("failed to update tradable pairs: %w", err)
+		}
+	}
+
+	a := b.GetAssetTypes(true)
+	var wg sync.WaitGroup
+	errC := make(chan error, len(a))
+	for i := range a {
+		wg.Add(1)
+		go func(a asset.Item) {
+			defer wg.Done()
+			if err := b.UpdateOrderExecutionLimits(ctx, a); err != nil && !errors.Is(err, common.ErrNotYetImplemented) {
+				errC <- fmt.Errorf("failed to set exchange order execution limits: %w", err)
+			}
+		}(a[i])
+	}
+	close(errC)
```

Comment:
This needs a wg.Wait() before it otherwise we'll panic on sending to a closed channel

Reply:
Yeah. That's a horrific oversight.
Must have been accidentally deleted, I'd assume.
Fixed d99caa20